ract/art/worm_simulator.py

```diff
@@ -0,0 +1,167 @@
+""" Move one lit voxel around the full cube. """
+
+import itertools
+import numpy as np
+from random import randint, shuffle
+
+from ract.interfaces import ArtworkRenderer
+from ract.utils import (
+    constant_frame, MAX_INTENSITY, MIN_INTENSITY, TimeStepper)
+
+
+class Worm(object):
+    UP = np.array([1, 0, 0])
+    DOWN = np.array([-1, 0, 0])
+    RIGHT = np.array([0, 1, 0])
+    LEFT = np.array([0, -1, 0])
+    IN = np.array([0, 0, 1])
+    OUT = np.array([0, 0, -1])
+    DIRECTIONS = [UP, DOWN, RIGHT, LEFT, IN, OUT]
+    WORM_INTENSITY = int(MAX_INTENSITY * 2 / 3)
+
+    def __init__(self, segment, direction, alive=True):
+        """ Create a worm
+        :param segment:
+        :param direction:
```

Comment:
Probably a bit pointless having these param bits if we're not going to fill them in (fine with them just being deleted though).